From 08ac02e2d9b0219cd3b7c7f622f09b07eff7bd26 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Wed, 25 Jan 2012 14:29:58 +0000
Subject: [PATCH 55/67] RHEL 6: Modify blkid test so it is successful on RHEL 6-era blkid.

---
 generator/actions.ml |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/generator/actions.ml b/generator/actions.ml
index e66ec73..862c8bb 100644
--- a/generator/actions.ml
+++ b/generator/actions.ml
@@ -8928,11 +8928,7 @@ device is stopped, but it is not destroyed or zeroed." };
       InitScratchFS, Always, TestOutputHashtable (
         [["blkid"; "/dev/sdb1"]],
         ["TYPE", "ext2";
-         "USAGE", "filesystem";
-         "PART_ENTRY_NUMBER", "1";
-         "PART_ENTRY_TYPE", "0x83";
-         "PART_ENTRY_OFFSET", "128";
-         "PART_ENTRY_SIZE", "102145"])
+         "USAGE", "filesystem"])
     ];
     shortdesc = "print block device attributes";
     longdesc = "\
-- 
1.7.1

