From a2614326d001f638750e8fee81cdd4b708c36928 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 25 Jan 2012 14:29:58 +0000
Subject: [PATCH 105/118] RHEL 6: Modify blkid test so it is successful on RHEL
 6-era blkid.

---
 generator/actions.ml | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/generator/actions.ml b/generator/actions.ml
index 5f1bd40..5b9a6b0 100644
--- a/generator/actions.ml
+++ b/generator/actions.ml
@@ -9000,11 +9000,7 @@ device is stopped, but it is not destroyed or zeroed." };
       InitScratchFS, Always, TestOutputHashtable (
         [["blkid"; "/dev/sdb1"]],
         ["TYPE", "ext2";
-         "USAGE", "filesystem";
-         "PART_ENTRY_NUMBER", "1";
-         "PART_ENTRY_TYPE", "0x83";
-         "PART_ENTRY_OFFSET", "128";
-         "PART_ENTRY_SIZE", "102145"])
+         "USAGE", "filesystem"])
     ];
     shortdesc = "print block device attributes";
     longdesc = "\
-- 
2.7.4

