From 671f45623c828120d066b18147b35b046a6a49dc Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 19 Sep 2012 15:02:27 +0100
Subject: [PATCH 107/118] RHEL 6: Exclude iptables from the appliance
 (RHBZ#858648).

---
 appliance/excludelist.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/appliance/excludelist.in b/appliance/excludelist.in
index aa4c406..c43006e 100644
--- a/appliance/excludelist.in
+++ b/appliance/excludelist.in
@@ -50,6 +50,9 @@
 ^redhat-logos
 ^dracut
 
+/* RHBZ#858648 */
+^iptables
+
 #endif
 
 #ifdef DEBIAN
-- 
2.7.4

